### PR TITLE
Changed to keep reading from UI in the event of a single failure #2480

### PR DESCRIPTION
**Pull request by myleshorton**
Some logging additions as well. Will flag the appropriate couple of lines.

### CODE REVIEW

File: src/github.com/getlantern/flashlight/ui/service.go [hunk 1]
```diff
@@ -140,15 +147,16 @@ func read() {
 		// Delegating response to the service that registered with the given type.
 		if services[envType.Type] == nil {
 			log.Errorf("Message type %v belongs to an unknown service.", envType.Type)
-			return
+			continue
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/ui/service.go]: This and the following are the only actual changes.

File: src/github.com/getlantern/flashlight/ui/service.go [hunk 2]
```diff
@@ -140,15 +147,16 @@ func read() {
...
 		}
 
 		env := &Envelope{}
 		err = json.Unmarshal(b, env)
 		if err != nil {
 			log.Errorf("Unable to unmarshal message of type %v: %v", envType.Type, err)
-			return
+			continue
```
Comment (myleshorton) [inline src/github.com/getlantern/flashlight/ui/service.go]: This too.

**Comment by myleshorton** [comment]
This at least makes things more robust if something odd happens. I suspect there's something else going on as well, but this should at least improve things.

**Comment by myleshorton** [comment]
This is a quick one @atavism if you want to have a peek. Hopefully it doesn't conflict with your other PR.

**Comment by atavism** [comment]
No, this shouldn't affect my changes. If we get an unknown message type (or can't decode a message), we keep the channel open anyway, assuming the service will register at a later point. Makes sense to me. Merging.

**Comment by myleshorton** [comment]
Cool thanks @atavism!

**Pull request merged**